paraswap/paraswap-volume-tracker
# feat: store and serve aggregated multichain stake data

store and serve raw details for each of the transactions for each of the stake chain, so that the aggregated amount can be computed from the data taken from the DB

Introduce new endpoint to serve that data: `/gas-refund/by-network/<address>/<fromEpoch>/<toEpoch>` with optional query parameter `showTransactions` that allows to inspect refunds details on per-transaction basis 

- refunded by staking and non staking chain chain
- claimable by staking chain
- inspect particular values on transaction-level

how to test:
1. run indexer 
2. fetch the endpoint
3. inspect some transactions to verify that the particulars out of which the stakeScore is computed look correct 
- owned PSP in sePSP2 at the moment of tx, sePSP1 balance, claimable sePSP1
- sum of `response.claimableByChain` equals sum of `response.refundedByChain`
- sum of `tx.claimableByStakeChain` for each of the tx equal `tx.refundedAmountPSP`
- whatever you can think out
 

# NB: before deploying 
We'll need to reindex all the previous epoch's data from scratch (until now) and update the prod DB

Here's a dump with indexed and "by stake network" data for epoch 38 (up to 7 sept): https://drive.google.com/file/d/1FszweSQpGzavzUSrNTHBM7Kv2lEY15EO/view?usp=sharing
